packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -449,10 +444,13 @@ class AnalyzeCommand extends FlutterCommand {
   Future<int> _analyzeWatch() async {
     List<String> directories;
 
-    if (isFlutterRepo) {
+    if (argResults['flutter-repo']) {
+      String root = path.absolute(ArtifactStore.flutterRoot);
+
       directories = <String>[];
-      directories.addAll(_gatherProjectPaths(path.absolute('examples')));
-      directories.addAll(_gatherProjectPaths(path.absolute('packages')));
+      directories.addAll(_gatherProjectPaths(path.join(root, 'examples')));
+      directories.addAll(_gatherProjectPaths(path.join(root, 'packages')));
+      directories.addAll(_gatherProjectPaths(path.join(root, 'dev')));
```

Comment:
Those files should be analyzed by --watch w/o us having to special case them. @pq has a change into the command-line analyzer that should allow it to analyze directories; that will help us unify our logic here.

It may be good to have a discussion about where we want to end up, in terms of what parts do what in this script, and what the user experience looks like from `dartanalyzer` / `flutter analyze` / Atom + analysis server based tools...